lib/snapshot.go

```diff
@@ -926,14 +906,14 @@ func (snap *Snapshot) DBSetAncestralRecordWithTxn(
 // AncestralRecordToDBEntry is used to translate the <ancestral_key, ancestral_value> pairs into
 // the actual <key, value> pairs. Ancestral records have the format:
 //
-//	<prefix [1]byte, block height [8]byte, key []byte> -> <value []byte, existence_byte [1]byte>
+//	<mainDbPrefix [1]byte, prefix [1]byte, block height [8]byte, key []byte> -> <value []byte, existence_byte [1]byte>
 //
 // So we need to trim the first 9 bytes off of the ancestral_key to get the actual key.
 // And we need to trim the last 1 byte off of the ancestral_value to get the actual value.
 func (snap *Snapshot) AncestralRecordToDBEntry(ancestralEntry *DBEntry) *DBEntry {
 	var dbKey, dbVal []byte
 	// Trim the prefix and the block height from the ancestral record key.
-	dbKey = ancestralEntry.Key[9:]
+	dbKey = ancestralEntry.Key[10:]
```

Comment:
Oooh tricky one...

Reply:
hahah yeah this one took me a minute to find.